django/utils/_os.py

```diff
@@ -56,18 +56,17 @@ def symlinks_supported():
     host platform and/or if they are allowed to be created (e.g.
     on Windows it requires admin permissions).
     """
-    tmpdir = tempfile.mkdtemp()
-    original_path = os.path.join(tmpdir, 'original')
-    symlink_path = os.path.join(tmpdir, 'symlink')
-    os.makedirs(original_path)
-    try:
-        os.symlink(original_path, symlink_path)
-        supported = True
-    except (OSError, NotImplementedError, AttributeError):
-        supported = False
-    else:
-        os.remove(symlink_path)
-    finally:
-        os.rmdir(original_path)
-        os.rmdir(tmpdir)
-        return supported
+    with tempfile.TemporaryDirectory() as tmpdir:
+        original_path = os.path.join(tmpdir, 'original')
+        symlink_path = os.path.join(tmpdir, 'symlink')
+        os.makedirs(original_path)
+        try:
+            os.symlink(original_path, symlink_path)
+            supported = True
+        except (OSError, NotImplementedError, AttributeError):
+            supported = False
+        else:
+            os.remove(symlink_path)
```

Comment:
Won't `symlink_path` and `original_path` be removed automatically as part of the cleanup? `tempfile.TemporaryDirectory` says "On completion of the context or destruction of the temporary directory object the newly created temporary directory and all its contents are removed from the filesystem."